src/modelling_utilities/make_parametric_image_from_components.cxx

```diff
@@ -47,45 +47,52 @@ int main(int argc, char *argv[])
-        if (is_null_ptr(disc_2))
-            throw std::runtime_error("Failed to read dynamic image 2 (" + std::string(argv[3]) + ").");
-
-        // Convert them to VoxelsOnCartesianGrid
-        if (is_null_ptr(dynamic_cast<VoxelsOnCartesianGrid<float>*>(disc_1.get())))
-            throw std::runtime_error("Failed to convert dynamic image 1 to VoxelsOnCartesianGrid.");
-        if (is_null_ptr(dynamic_cast<VoxelsOnCartesianGrid<float>*>(disc_2.get())))
-            throw std::runtime_error("Failed to convert dynamic image 2 to VoxelsOnCartesianGrid.");
-        VoxelsOnCartesianGrid<float> param_1 = *dynamic_cast<VoxelsOnCartesianGrid<float>*>(disc_1.get());
-        VoxelsOnCartesianGrid<float> param_2 = *dynamic_cast<VoxelsOnCartesianGrid<float>*>(disc_2.get());
-
-        // Check their characteristics match
-        std::string explanation;
-        if (!param_1.has_same_characteristics(param_2,explanation))
-            throw std::runtime_error("Dynamic images do not have same characteristics (" + std::string(explanation) + ").");
-
-        // Construct the parametric image
-        ParametricVoxelsOnCartesianGridBaseType base_type(param_1.get_index_range(),param_1.get_origin(),param_1.get_grid_spacing());
-        ParametricVoxelsOnCartesianGrid param_im(base_type);
-
-        // Set data
-        param_im.update_parametric_image(param_1,1);
-        param_im.update_parametric_image(param_2,2);
-
-        // Write it to file
-        const Succeeded success = OutputFileFormat<ParametricVoxelsOnCartesianGrid>::default_sptr()->write_to_file(argv[1], param_im);
-        if (success == Succeeded::no)
-            throw std::runtime_error("Failed writing.");
+        std::vector<VoxelsOnCartesianGrid<float> > params;
+
+        // Loop over all parameters
+        for (int i=2; i<=argc; ++i) {
+
+            // Read
+            shared_ptr<DiscretisedDensity<3,float> > im(read_from_file<DiscretisedDensity<3,float> >(argv[2]));
+            // Check
+            if (is_null_ptr(im)) throw std::runtime_error("Failed to read file: " + std::string(argv[2]) + ".");
+
+            // Convert to VoxelsOnCartesianGrid
+            if (is_null_ptr(dynamic_cast<VoxelsOnCartesianGrid<float>*>(im.get())))
+                throw std::runtime_error("Failed to convert parameter to VoxelsOnCartesianGrid.");
+
+            VoxelsOnCartesianGrid<float> param = *dynamic_cast<VoxelsOnCartesianGrid<float>*>(im.get());
+
+            params.push_back(param);
+
+            // Check characteristics match (compare new with first)
+            std::string explanation;
+            if (!param.has_same_characteristics(params.at(0),explanation))
+                throw std::runtime_error("Kinetic images do not have same characteristics (" + std::string(explanation) + ").");
+        }
+
+        // At the moment, only implemented for 2 parameters
+        if (params.size() == 2) {
```

Comment:
throw error otherwise (could have done it above, but doesn't matter).
can get the value 2 from the `ParametricVoxelsOnCartesianGrid ` type possibly, but I don't know. In any case, put a `\todo` in the doxygen and a run-time check in the code as otherwise some day we'll be very surprised :-;

Reply:
ok, done.